hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSink.java

```diff
@@ -236,81 +258,71 @@ public void replicateEntries(List<WALEntry> entries, final ExtendedCellScanner c
             throw new ArrayIndexOutOfBoundsException("Expected=" + count + ", index=" + i);
           }
           ExtendedCell cell = cells.current();
-          // Handle bulk load hfiles replication
           if (CellUtil.matchingQualifier(cell, WALEdit.BULK_LOAD)) {
+            // Bulk load events
             BulkLoadDescriptor bld = WALEdit.getBulkLoadDescriptor(cell);
             if (bld.getReplicate()) {
-              if (bulkLoadsPerClusters == null) {
-                bulkLoadsPerClusters = new HashMap<>();
-              }
-              // Map of table name Vs list of pair of family and list of
-              // hfile paths from its namespace
+              // Map of tableNameStr to (family, hfile paths) pairs
               Map<String, List<Pair<byte[], List<String>>>> bulkLoadHFileMap =
                 bulkLoadsPerClusters.computeIfAbsent(bld.getClusterIdsList(), k -> new HashMap<>());
-              buildBulkLoadHFileMap(bulkLoadHFileMap, table, bld);
+              buildBulkLoadHFileMap(bulkLoadHFileMap, bld);
             }
           } else if (CellUtil.matchingQualifier(cell, WALEdit.REPLICATION_MARKER)) {
             Mutation put = processReplicationMarkerEntry(cell);
             if (put == null) {
               continue;
             }
-            table = REPLICATION_SINK_TRACKER_TABLE_NAME;
-            List<UUID> clusterIds = new ArrayList<>();
-            for (HBaseProtos.UUID clusterId : entry.getKey().getClusterIdsList()) {
-              clusterIds.add(toUUID(clusterId));
-            }
+            List<UUID> clusterIds = getSourceClusterIds(entry);
             put.setClusterIds(clusterIds);
-            addToHashMultiMap(rowMap, table, clusterIds, put);
+            addToHashMultiMap(sinkRowMap, REPLICATION_SINK_TRACKER_TABLE_NAME, clusterIds, put);
           } else {
-            // Handle wal replication
-            if (isNewRowOrType(previousCell, cell)) {
-              // Create new mutation
-              mutation = CellUtil.isDelete(cell)
-                ? new Delete(cell.getRowArray(), cell.getRowOffset(), cell.getRowLength())
-                : new Put(cell.getRowArray(), cell.getRowOffset(), cell.getRowLength());
-              List<UUID> clusterIds = new ArrayList<>(entry.getKey().getClusterIdsList().size());
-              for (HBaseProtos.UUID clusterId : entry.getKey().getClusterIdsList()) {
-                clusterIds.add(toUUID(clusterId));
-              }
-              mutation.setClusterIds(clusterIds);
-              mutation.setAttribute(ReplicationUtils.REPLICATION_ATTR_NAME,
+            TableName sinkTableName = translator.getSinkTableName(tableName);
+            ExtendedCell sinkCell = translator.getSinkExtendedCell(tableName, cell);
```

Comment:
This is the reason why I want to see the javadoc for this method,as why do we need to pass the original table name in? And I think we will just do tableName mapping, so we do not need to call the above getSinkTableName everytime as all the cells from the WALEntry are for the same table?